### ISSUE TITLE
Training and Validation Sets

### ISSUE THREAD

**Issue by alex-hh** [vam-sin/domdet#4]
Thought I'd use an issue just to record some things we might need to resolve.

## CATH Validation / Test Sets

We can use CATH topology codes to ensure no domain in the training set has any structural overlap to any domain in the validation/test set, similar to the standard practice in inverse folding (Ingraham, ESM-IF). We can make use of the existing ESM-IF splits if we like, however the train split has limited multi domain proteins, and imposes quite strong redundancy filtering at the level of domains (as a consequence of the only chains considered for the set being chains having domains in the 40% sequence identity non-redundant set provided by CATH).

We should probably split the validation set into different categories by number of domains
- see the CASP evaluations paper - we know that single domains will be very easy and might dominate the scores.

## Training Set

As soon as we have a fixed validation set, we can iterate on choices of training set (while imposing the no topology overlap criterion). We might choose to:
* directly use anything in our current dataset which doesn't have topology overlap to the validation/test sets
* directly use the Facebook training set (although a reasonable fraction of these are missing)
* impose some redundancy filtering (e.g. at the domain combinations level - each combination of domains at some level of redundancy control (e.g. S60) should occur only once).
* use any of the above, down weighting/sampling single domain proteins which may otherwise dominate

Ultimately validation performance can decide which of these is best.

## Cropping

We need to make sure that during validation the correct (full) chain is used

Couple of possibilities for what we do during training:
 * random crops of fixed size
 * crops of fixed size preserving entire domains (i.e. avoid cropping the middle of a domain to avoid cases like we saw on pymol where the inputs are insufficient to determine the labels due to missing chunks of the domain).
* variable size versions of either of the above, by adding a length-based batch sampler
* no cropping at all, by adding a length-based batch sampler
